File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    vector<int> ST;
    int N = board[0].size();
    int doll = 0;
    
    
    for(int move : moves){
        for(int depth = 0 ; depth < N ; depth++){
            if(board[depth][move-1]){
                doll = board[depth][move-1];
                ST.push_back(doll);
                if(ST.size() >= 2 && ST[ST.size()-2] == doll){
                    ST.pop_back();
                    ST.pop_back();
                    answer += 2;
                }
                board[depth][move-1] = 0;
                break;
            }
        }
    }
    
    
    return answer;
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    ST = [] #바구니
    N = len(board[0])
    
    #인형을 뺀다.
    for move in moves:
        for depth in range(N):
            if(board[depth][move-1]):
                doll=board[depth][move-1] #인형의 번호
                ST.append(doll)  #바구니에 넣는다
                if(len(ST)>=2 and ST[-2] == doll):  #하나 넣을때마다 터지는게 있나 검사를 한다.
                    ST.pop()
                    ST.pop()
                    answer += 2
                board[depth][move-1] = 0  #보드의 인형이 있던 자리를 0으로 바꾼다
                break
            
    
    return answer